lib/feedback_api_web/facades/survey_create_facade.ex

```diff
@@ -0,0 +1,65 @@
+defmodule FeedbackApiWeb.SurveyCreateFacade do
+  alias FeedbackApi.{Survey, User, Question, Group, Repo}
+  import Ecto.Query
+
+  def create_survey(arguments) do
+    Repo.transaction(fn ->
+      try do
+        survey = Survey.changeset(%Survey{}, arguments) |> Repo.insert!()
+        create_groups(survey, arguments["groups"])
+        create_questions(survey, arguments["questions"])
+      rescue
+        e -> Repo.rollback("Missing required fields")
+      end
+    end)
+  end
+
+  defp create_groups(survey, groups) do
+    Enum.map(groups, fn group -> create_group(survey, group) end)
+  end
+
+  defp create_questions(survey, questions) do
+    Enum.map(questions, fn question -> create_question(survey, question) end)
+  end
+
+  defp create_group(survey, group) do
+    new_group =
+      Group.changeset(%Group{}, group) |> Repo.insert!() |> Repo.preload([:survey, :users])
+
+    users = Repo.all(from u in User, where: u.id in ^group["members_ids"])
+
+    group_associations_changeset =
+      Ecto.Changeset.change(new_group)
+      |> Ecto.Changeset.put_assoc(:survey, survey)
```

Comment:
Just to make sure--is this just declaring the associations explicitly? And you have to do that in a separate changeset?

Reply:
As far as I know, yes. You cannot directly assign foreign key values, and must include an association changeset, which is then committed.